internal/transform/utils.go

```diff
@@ -15,3 +17,37 @@ func hasTruthyAttr(n *astro.Node, key string) bool {
 	}
 	return false
 }
+
+func HasAttr(n *astro.Node, key string) bool {
+	for _, attr := range n.Attr {
+		if attr.Key == key {
+			return true
+		}
+	}
+	return false
+}
+
+func GetAttrs(n *astro.Node) js.Value {
+	attrs := js.Global().Get("Object").New()
+	for _, attr := range n.Attr {
+		switch attr.Type {
+		case astro.QuotedAttribute:
+			attrs.Set(attr.Key, attr.Val)
+		case astro.EmptyAttribute:
+			attrs.Set(attr.Key, true)
+		}
+	}
+	return attrs
+}
```

Comment:
This function returns compile-time known attributes as a JS object, which can then be passed to the preprocessors.